stdlib/public/core/RuntimeFunctionCounters.swift

```diff
@@ -0,0 +1,506 @@
+      result[i] = other[i] - self[i]
+    }
+    return result
+  }
+}
+
+/// This type should be used to collect statistics about the global runtime
+/// function pointers.
+public // @testable
+struct _GlobalRuntimeFunctionCountersState: _RuntimeFunctionCountersStats {
+  var state = _RuntimeFunctionCountersState()
+
+  public init() {
+    getGlobalRuntimeFunctionCounters()
+  }
+
+  mutating public func getGlobalRuntimeFunctionCounters() {
+    _RuntimeFunctionCounters.getGlobalRuntimeFunctionCounters(&state)
+  }
+
+  mutating public func setGlobalRuntimeFunctionCounters() {
+    _RuntimeFunctionCounters.setGlobalRuntimeFunctionCounters(&state)
+  }
+
+  public subscript(_ index: String) -> UInt32 {
+    get {
+      return state[index]
+    }
+    set {
+      state[index] = newValue
+    }
+  }
+
+  public subscript(_ index: Int) -> UInt32 {
+    get {
+      return state[index]
+    }
+    set {
+      state[index] = newValue
+    }
+  }
+}
+
+/// This type should be used to collect statistics about object runtime
+/// function pointers.
+public // @testable
+struct _ObjectRuntimeFunctionCountersState: _RuntimeFunctionCountersStats {
+  var state = _RuntimeFunctionCountersState()
+
+  // Initialize with the counters for a given object.
+  public init(_ p: UnsafeRawPointer) {
+    getObjectRuntimeFunctionCounters(p)
+  }
+
+  public init() {
+  }
+
+  mutating public func getObjectRuntimeFunctionCounters(_ o: UnsafeRawPointer) {
+    _RuntimeFunctionCounters.getObjectRuntimeFunctionCounters(o, &state)
+  }
+
+  mutating public func setObjectRuntimeFunctionCounters(_ o: UnsafeRawPointer) {
+    _RuntimeFunctionCounters.setObjectRuntimeFunctionCounters(o, &state)
+  }
+
+  public subscript(_ index: String) -> UInt32 {
+    get {
+      return state[index]
+    }
+    set {
+      state[index] = newValue
+    }
+  }
+
+  public subscript(_ index: Int) -> UInt32 {
+    get {
+      return state[index]
+    }
+    set {
+      state[index] = newValue
+    }
+  }
+}
+
+/// Collects all references inside an object.
+/// Runtime counters tracking is disabled for the duration of this operation
+/// so that it does not affect those counters.
+public // @testable
+func _collectReferencesInsideObject(_ value: Any) -> [UnsafeRawPointer] {
+  let savedMode = _RuntimeFunctionCounters.disableRuntimeFunctionCountersUpdates()
+  // Collect all references inside the object
+  let refs = _RuntimeFunctionCounters.collectAllReferencesInsideObject(value)
+  _RuntimeFunctionCounters.enableRuntimeFunctionCountersUpdates(mode: savedMode)
+  return refs
+}
+
+/// A helper method to measure how global and per-object function counters
+/// were changed during execution of a closure provided as a parameter.
+/// Returns counter diffs for global counters and for the object-specific
+/// counters related to a given object.
+public // @testable
+func _measureRuntimeFunctionCountersDiffs(
+  objects: [UnsafeRawPointer], _ f: () -> Void) ->
```

Comment:
`f` is not very descriptive. `body` perhaps? At least it has some part of the overall meaning...